boot/boot_serial/src/boot_serial.c

```diff
@@ -275,7 +275,7 @@ bs_list(char *buf, int len)
         int swap_status = boot_swap_type_multi(image_index);
 #endif
 
-        for (slot = 0; slot < MCUBOOT_IMAGE_NUMBER; slot++) {
+        for (slot = 0; slot < BOOT_NUM_SLOTS; slot++) {
```

Comment:
This is proper fix.
I just checked that BOOT_NUM_SLOTS is defined to be `2`. Looks like this is not true if we came to single bank.